.github/actions/generate-commit-list-v1/src/getParsedCommitList.ts

```diff
@@ -0,0 +1,37 @@
+import getCommitType from './getCommitType'
+import { GetParsedCommitListParams, ParsedCommitList } from './types'
+
+/**
+ * Parses the raw commit list into an array of commit objects.
+ * @param {object} GetParsedCommitListParams Raw commit list and repository URL.
+ * @returns An array of parsed commits.
+ */
+
+export default function getParsedCommitList({
+  rawCommitList,
+  repository
+}: GetParsedCommitListParams): Array<ParsedCommitList> {
+  const parsedCommits: Array<ParsedCommitList> = []
+
+  for (const commit of rawCommitList) {
+    // group sha and title together
+    //eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const shaAndTitle = commit.match(/^(.{7}) (.+?)(?:\s\(#\d+\))?$/)!
```

Comment:
This assumes that the sha is only 7 characters long, which is not always the case. [Git docs](https://git-scm.com/book/en/v2/Git-Tools-Revision-Selection#Short-SHA-1) says it grows as the repo commits do 

> Generally, eight to ten characters are more than enough to be unique within a project. For example, as of February 2019, the Linux kernel (which is a fairly sizable project) has over 875,000 commits and almost seven million objects in its object database, with no two objects whose SHA-1s are identical in the first 12 characters.

We'll also probably want to scope the sha to valid hex digits rather than accepting everything

```suggestion
    const shaAndTitle = commit.match(/^([0-9a-f]+) (.+?)(?:\s\(#\d+\))?$/)!
```

Also some additional test cases we'll want:

- `1c41b7a1 revert: feat: awesome feature (#3957) (#3988)` 
- `15a29139 chore(release): 4.8.1`